core/server/api_container/server/startosis_engine/kurtosis_types/service_config/ready_conditions.go

```diff
@@ -0,0 +1,213 @@
+		KurtosisBaseBuiltin: &kurtosis_starlark_framework.KurtosisBaseBuiltin{
+			Name: ReadyConditionsTypeName,
+			Arguments: []*builtin_argument.BuiltinArgument{
+				{
+					Name:              RecipeAttr,
+					IsOptional:        false,
+					ZeroValueProvider: builtin_argument.ZeroValueProvider[starlark.Value],
+					Validator:         nil,
+				},
+				{
+					Name:              FieldAttr,
+					IsOptional:        false,
+					ZeroValueProvider: builtin_argument.ZeroValueProvider[starlark.String],
+					Validator:         nil,
+				},
+				{
+					Name:              AssertionAttr,
+					IsOptional:        false,
+					ZeroValueProvider: builtin_argument.ZeroValueProvider[starlark.String],
+					Validator:         assert.ValidateAssertionToken,
+				},
+				{
+					Name:              TargetAttr,
+					IsOptional:        false,
+					ZeroValueProvider: builtin_argument.ZeroValueProvider[starlark.Comparable],
+					Validator:         nil,
+				},
+				{
+					Name:              IntervalAttr,
+					IsOptional:        true,
+					ZeroValueProvider: builtin_argument.ZeroValueProvider[starlark.String],
+					Validator:         nil,
+				},
+				{
+					Name:              TimeoutAttr,
+					IsOptional:        true,
+					ZeroValueProvider: builtin_argument.ZeroValueProvider[starlark.String],
+					Validator:         nil,
+				},
+			},
+		},
+		Instantiate: instantiateReadyConditions,
+	}
+}
+
+func instantiateReadyConditions(arguments *builtin_argument.ArgumentValuesSet) (kurtosis_type_constructor.KurtosisValueType, *startosis_errors.InterpretationError) {
+	kurtosisValueType, err := kurtosis_type_constructor.CreateKurtosisStarlarkTypeDefault(ReadyConditionsTypeName, arguments)
+	if err != nil {
+		return nil, err
+	}
+	return &ReadyConditions{
+		KurtosisValueTypeDefault: kurtosisValueType,
+		recipe:                   nil,
+		field:                    "",
+		assertion:                "",
+		target:                   starlark.String(""),
+		interval:                 defaultInterval,
+		timeout:                  defaultTimeout,
+	}, nil
+}
+
+// ReadyConditions is a starlark.Value that holds all the information needed for ensuring service readiness
+type ReadyConditions struct {
+	*kurtosis_type_constructor.KurtosisValueTypeDefault
+	recipe    recipe.Recipe
+	field     string
+	assertion string
+	target    starlark.Comparable
+	interval  time.Duration
+	timeout   time.Duration
+}
+
+func (readyConditions *ReadyConditions) ValidateAndFillReadyConditions() *startosis_errors.InterpretationError {
+
+	var (
+		genericRecipe     recipe.Recipe
+		found             bool
+		httpRecipe        *recipe.HttpRequestRecipe
+		execRecipe        *recipe.ExecRecipe
+		interpretationErr *startosis_errors.InterpretationError
+	)
+
+	httpRecipe, found, interpretationErr = kurtosis_type_constructor.ExtractAttrValue[*recipe.HttpRequestRecipe](readyConditions.KurtosisValueTypeDefault, RecipeAttr)
+	genericRecipe = httpRecipe
+	if interpretationErr != nil || !found {
```

Comment:
`interpretationErr  != nil` is really different than `found` here. If `found == true`, it's pointless to try to parse it as a ExecRecipe as the argument simply doesn't exist at all. I think this would improve readability:
```
if !found {
    return startosis_errors.NewInterpretationError("Required attribute '%s' could not be found on type '%s'",
				RecipeAttr, ReadyConditionsTypeName)
}
if interpretationErr != nil {
    ... try parse it as an ExecRecipe
}
```